tests/player.c

```diff
@@ -536,6 +439,335 @@ START_TEST (test_play_audio_eos)
 
 END_TEST;
 
+static void
+test_audio_info (GstPlayerMediaInfo * media_info)
+{
+  gint i = 0;
+  GList *list;
+
+  for (list = gst_player_get_audio_streams (media_info);
+      list != NULL; list = list->next) {
+    GstPlayerStreamInfo *stream = (GstPlayerStreamInfo *) list->data;
+    GstPlayerAudioInfo *audio_info = (GstPlayerAudioInfo *) stream;
+
+    fail_unless (gst_player_stream_info_get_tags (stream) != NULL);
+    fail_unless (gst_player_stream_info_get_caps (stream) != NULL);
+    fail_unless_equals_string (
+        gst_player_stream_info_get_stream_type (stream), "audio");
+
+    if (i == 0) {
+      fail_unless_equals_string (gst_player_stream_info_get_codec (stream),
+          "MPEG-1 Layer 3 (MP3)");
+      fail_unless_equals_int (gst_player_audio_info_get_sample_rate
+          (audio_info), 48000);
+      fail_unless_equals_int (gst_player_audio_info_get_channels (audio_info),
+          2);
+      fail_unless_equals_int (gst_player_audio_info_get_max_bitrate
+          (audio_info), 192000);
+      fail_unless (gst_player_audio_info_get_language (audio_info) != NULL);
+    } else {
+      fail_unless_equals_string (gst_player_stream_info_get_codec (stream),
+          "MPEG-4 AAC");
+      fail_unless_equals_int (gst_player_audio_info_get_sample_rate
+          (audio_info), 48000);
+      fail_unless_equals_int (gst_player_audio_info_get_channels (audio_info),
+          6);
+      fail_unless (gst_player_audio_info_get_language (audio_info) != NULL);
+    }
+
+    i++;
+  }
+}
+
+static void
+test_video_info (GstPlayerMediaInfo * media_info)
+{
+  GList *list;
+
+  for (list = gst_player_get_video_streams (media_info);
+      list != NULL; list = list->next) {
+    gint fps_d, fps_n;
+    guint par_d, par_n;
+    GstPlayerStreamInfo *stream = (GstPlayerStreamInfo *) list->data;
+    GstPlayerVideoInfo *video_info = (GstPlayerVideoInfo *) stream;
+
+    fail_unless (gst_player_stream_info_get_tags (stream) != NULL);
+    fail_unless (gst_player_stream_info_get_caps (stream) != NULL);
+    fail_unless_equals_int (gst_player_stream_info_get_index (stream), 0);
+    fail_unless_equals_string (gst_player_stream_info_get_codec (stream),
+        "H264");
```

Comment:
Fails with latest GStreamer GIT:
player.c:498:F:general:test_play_media_info:0: 'gst_player_stream_info_get_codec (stream)' (H.264 (High Profile)) is not equal to '"H264"' (H264)